crates/node/src/lib.rs

```diff
@@ -149,6 +149,22 @@ impl Node {
         let is_provisioned =
             cluster_marker::validate_and_update_cluster_marker(config.common.cluster_name())?;
 
+        if !matches!(
+            &config.common.metadata_client.kind,
+            restate_types::config::MetadataClientKind::Replicated { .. } // used for both local and remote Restate metadata store clients
```

Comment:
Technically, the local metadata server uses an in memory client now and ignores the configured `MetadataClientKind`.

If `MetadataServerKind::Local` and `Role::MetadataServer` is configured, then we are ignoring any other `MetadataStoreClient` which can be confusing. I guess this happens if you leave the metadata server role configured and want to use an external metadata store by only configuring the metadata client options.

If `MetadataServerKind::Replicated` and `Role::MetadataServer` and not `MetadataClientKind::Replicated`, then we are kind of ignoring the replicated metadata server and using whatever client was configured (etcd, object-store, etc.).

Once we remove the local metadata server, this asymmetry is hopefully gone.

Reply:
I tried to capture this in a comment - you be the judge of whether I succeeded :-)